docs/db-migration-guide.ipynb

```diff
@@ -0,0 +1,137 @@
+    "juju remove-relation katib-db-manager katib-db\n",
+    "\n",
+    "# connect to unit's IP address using mysql client and verify that all data is intact\n",
+    "mysql --host=$KATIB_DB_IP --user=$KATIB_DB_USER --password=$KATIB_DB_PASSWORD\n",
+    "\n",
+    "# create backup of all databases file using `mysqldump` utility, username, password, and unit's IP address\n",
+    "# onbtained earlier\n",
+    "mysqldump --host=$KATIB_DB_IP --user=$KATIB_DB_USER --password=$KATIB_DB_PASSWORD --column-statistics=0 --databases katib  > katib-db.sql\n",
+    "\n",
+    "# deploy new MySQL database charm\n",
+    "juju deploy mysql-k8s katib-db-mysql --channel 8.0/stable --trust\n",
+    "\n",
+    "# obtain username and password of new MySQL database from DB relation\n",
+    "# note that since relation is not yet established it is impossible to retrieve this information from that relation\n",
+    "# the relation can only be established after data is retored into new database\n",
+    "MYSQL_DB_USER=$(juju run-action katib-db-mysql/0 get-password --wait | yq '.[] | .results.username')\n",
+    "MYSQL_DB_PASSWORD=$(juju run-action katib-db-mysql/0 get-password --wait | yq '.[] | .results.password')\n",
+    "MYSQL_DB_IP=$(juju show-unit katib-db/0 | yq '.[] | .address')\n",
+    "\n",
+    "# connect to new DB using username, password, and unit's IP address and restore database from backup\n",
+    "mysql --host=$MYSQL_DB_IP --user=$MYSQL_DB_USER --password=$MYSQL_DB_PASSWORD < katib-db.sql\n",
+    "\n",
+    "# relate katib-db-manager and new MySQL database charm\n",
+    "juju relate katib-db-manager:relational-db katib-db-mysql:database"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## KFP DB migration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# obtain username and password of existing MariadDB database from DB relation\n",
+    "# in Charmed Kubeflow 1.6 and 1.7 username used is `root` and password is specified in `mysql` relation by\n",
+    "# 'root_password'\n",
+    "KFP_DB_MYSQL_RELATION_ID=$(juju show-unit kfp-db/0 | yq '.[] | .relation-info | select(.[].endpoint == \"mysql\") | .[0] | .relation-id')\n",
+    "KFP_DB_USER=root\n",
+    "KFP_DB_PASSWORD=$(bash -c \"juju run --unit kfp-db/0 'relation-get -r $KFP_DB_MYSQL_RELATION_ID - kfp-db/0' | grep root_password\" | awk '{print $2}')\n",
+    "KFP_DB_IP=$(juju show-unit kfp-db/0 | yq '.[] | .address')\n",
+    "\n",
+    "# ensure that there are no new connections are made and that dababase is not altered\n",
+    "# remove relation between kfp-api charm and MariaDB charm\n",
+    "juju remove-relation kfp-api kfp-db\n",
+    "\n",
+    "# connect to unit's IP address using mysql client and verify that all data is intact\n",
```

Comment:
How would users know data is intact?

Reply:
Will add.

Reply:
Updated.

Reply:
I think this still applies

Reply:
Should provide SQL queries? How deep should we go into describing DB operations?
I'd would leave it outside of this guide, but open to suggestions.

Reply:
I commented because you had it initially, if you don't think it is required we are good